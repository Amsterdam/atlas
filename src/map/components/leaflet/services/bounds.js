export const getBounds = (element) => {
  // if activeElement is a shape
  if (element.getBounds) {
    const elementBounds = element.getBounds();
    if (Object.keys(elementBounds).length) {
      return elementBounds;
    }
  // if activeElement is a point
  } else {
    const latLng = element.getLatLng();
    return [
      [latLng.lat, latLng.lng],
      [latLng.lat, latLng.lng]
    ];
  }
  return [];
};

export const boundsToString = (elementBounds) => (
  elementBounds.toBBoxString ?
    elementBounds.toBBoxString() : elementBounds.toString());
