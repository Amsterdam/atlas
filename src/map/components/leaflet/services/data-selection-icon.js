import { DivIcon } from 'leaflet';

import ICON_CONFIG from './icon-config.constant';

const dataSelectionIcon = (count) => new DivIcon({
  ...ICON_CONFIG.DATA_SELECTION,
  popupAnchor: null,
  className: 'o-highlight-cluster',
  html: `
    <div
      aria-label="Cluster met ${count} appartementen"
      class="o-highlight-cluster__text"
    >
      ${count}
    </div>
  `
});

export default dataSelectionIcon;
