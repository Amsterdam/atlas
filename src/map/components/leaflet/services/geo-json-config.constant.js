const geoJsonConfig = {
  dataSelection: {
    style: {
      color: '#FF0000'
    }
  },
  dataSelectionAlternate: {
    style: {
      color: '#0000FF',
      dashArray: '10 10'
    }
  }
};

export default geoJsonConfig;
export const dataSelectionType = 'dataSelection';
export const dataSelectionAlternateType = 'dataSelectionAlternate';
