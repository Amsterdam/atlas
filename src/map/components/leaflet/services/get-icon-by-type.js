import searchIcon from './search-icon';
import { straatbeeldPersonIcon, straatbeeldOrientationIcon } from './straatbeeld-icon';

export const geoSearch = 'straatbeeldPerson';
export const straatbeeldPerson = 'straatbeeldPerson';
export const straatbeeldOrientation = 'straatbeeldOrientation';

const getIconByType = (type) => {
  switch (type) {
    case straatbeeldPerson:
      return straatbeeldPersonIcon;

    case straatbeeldOrientation:
      return straatbeeldOrientationIcon;

    default:
      return searchIcon;
  }
};

export default getIconByType;
