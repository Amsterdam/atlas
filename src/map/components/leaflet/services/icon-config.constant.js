const ICON_CONFIG = {
  SEARCH: {
    iconUrl: 'assets/images/map/search.svg',
    iconSize: [48, 77],
    iconAnchor: [23, 65]
  },
  DETAIL: {
    iconUrl: 'assets/images/map/detail.svg',
    iconSize: [21, 21],
    iconAnchor: [10, 10]
  },
  DATA_SELECTION: {
    iconSize: [39, 39],
    iconAnchor: [19, 19]
  },
  STRAATBEELD_ORIENTATION: {
    iconUrl: 'assets/images/map/straatbeeld-orientation.svg',
    iconSize: [70, 70],
    iconAnchor: [35, 35]
  },
  STRAATBEELD_PERSON: {
    iconUrl: 'assets/images/map/straatbeeld-person.svg',
    iconSize: [18, 31],
    iconAnchor: [9, 22]
  }
};

export default ICON_CONFIG;
