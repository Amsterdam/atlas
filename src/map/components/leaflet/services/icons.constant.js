import searchIcon from './search-icon';
import { panoramaPersonIcon, panoramaOrientationIcon } from './panorama-icon';
import detailIcon from './detail-icon';
import dataSelectionIcon from './data-selection-icon';
import locationIcon from './location-icon';

export const geoSearchType = 'geoSearchType';
export const dataSelectionType = 'dataSelectionType';
export const detailPointType = 'detailPointType';
export const panoramaPersonType = 'panoramaPersonType';
export const panoramaOrientationType = 'panoramaOrientationType';
export const markerPointType = 'markerPointType';

const icons = {
  [geoSearchType]: searchIcon,
  [dataSelectionType]: dataSelectionIcon,
  [detailPointType]: detailIcon,
  [panoramaPersonType]: panoramaPersonIcon,
  [panoramaOrientationType]: panoramaOrientationIcon,
  [markerPointType]: locationIcon
};
export default icons;
