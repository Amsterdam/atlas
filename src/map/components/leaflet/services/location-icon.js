import { Icon } from 'leaflet'

import ICON_CONFIG from './icon-config.constant'

const locationIcon = type =>
  new Icon({
    ...(ICON_CONFIG.LOCATION_MARKER[type]
      ? ICON_CONFIG.LOCATION_MARKER[type]
      : ICON_CONFIG.LOCATION_MARKER.default),
    popupAnchor: null,
    shadowUrl: null,
    shadowSize: null,
    shadowAnchor: null,
    className: 'dp-leaflet-location-icon',
  })

export default locationIcon
