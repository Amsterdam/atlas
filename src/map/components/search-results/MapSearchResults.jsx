import React from 'react';
import PropTypes from 'prop-types';

import { wgs84ToRd } from '../../../shared/services/coordinate-reference-system';
import MapSearchResultsCategory from './MapSearchResultsCategory';
import Notification from '../../../shared/components/notification/Notification';
import MaximizeIcon from '../../../../public/images/icon-maximize.svg';

export const previewPanelResultLimit = 3;

const MapSearchResults = ({
  count,
  location,
  missingLayers,
  onItemClick,
  panoUrl,
  results,
  onMaximize,
  onPanoPreviewClick
}) => {
  const rdCoordinates = wgs84ToRd(location);

  return (
    <section className="map-search-results">
      <header
        className={`
          map-search-results__header
          map-search-results__header--${panoUrl ? 'pano' : 'no-pano'}
        `}
      >
        <button
          className="map-search-results__header-pano-button"
          onClick={onPanoPreviewClick}
          title="Panoramabeeld"
        >
          {panoUrl && (
            <img
              alt="Panoramabeeld"
              width="438"
              height="292"
              className="map-search-results__header-pano"
              src={panoUrl}
            />
          )}
          <div className="map-search-results__header-container">
            <h1 className="map-search-results__header-title">Resultaten ({count})</h1>
            <h2 className="map-search-results__header-subtitle">
              locatie {rdCoordinates.x.toFixed(2)}, {rdCoordinates.y.toFixed(2)}
            </h2>
          </div>
        </button>
      </header>
      <div className="map-search-results__scroll-wrapper">
        <ul className="map-search-results__list">
          {missingLayers && (
            <li>
              <Notification>Geen details beschikbaar van: {missingLayers}</Notification>
            </li>
          )}
          {
            results.map((mainCategory) => (
              <MapSearchResultsCategory
                key={mainCategory.categoryLabel}
                category={mainCategory}
                onClick={onItemClick}
              />
            ))
          }
        </ul>
        <footer className="map-search-results__footer">
          <button onClick={onMaximize} className="map-search-results__button">
            <MaximizeIcon className="map-search-results__button-icon" />
            Volledig weergeven
          </button>
        </footer>
      </div>
    </section>
  );
};

MapSearchResults.propTypes = {
  count: PropTypes.number, // eslint-disable-line
  location: PropTypes.object, // eslint-disable-line
  missingLayers: PropTypes.string, // eslint-disable-line
  onItemClick: PropTypes.func.isRequired,
  onMaximize: PropTypes.func.isRequired,
  onPanoPreviewClick: PropTypes.func.isRequired,
  panoUrl: PropTypes.string, // eslint-disable-line
  results: PropTypes.array // eslint-disable-line
};

export default MapSearchResults;
