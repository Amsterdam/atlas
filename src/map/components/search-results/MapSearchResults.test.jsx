import { shallow } from 'enzyme'

import MapSearchResults from './MapSearchResults'
import { wgs84ToRd } from '../../../shared/services/coordinate-reference-system'

jest.mock('redux-first-router-link', () => ({
  _esModule: true,
  default: jest.fn,
}))

jest.mock('../../../shared/services/coordinate-reference-system')
jest.mock('react-redux', () => ({
  useSelector: jest.fn(() => []),
}))

jest.mock('../../../app/utils/useGetLegacyPanoramaPreview', () => () => ({
  panoramaUrl: '/foo',
  link: '/link',
  linkComponent: null,
}))

describe('MapSearchResults', () => {
  beforeEach(() => {
    wgs84ToRd.mockImplementation(() => ({ x: 1.12345, y: 654.321 }))
  })

  afterEach(() => {
    wgs84ToRd.mockReset()
  })

  it('should calculate RD coordinates', () => {
    const location = {
      latitude: 15.999,
      longitude: 329.123,
    }
    const clickHandler = jest.fn()
    shallow(
      <MapSearchResults
        isEmbed={false}
        location={location}
        onItemClick={clickHandler}
        onMaximize={clickHandler}
        results={[]}
      />,
    )
    expect(wgs84ToRd).toHaveBeenCalledWith(location)
  })

  describe('rendering', () => {
    it('should render empty results', () => {
      const clickHandler = jest.fn()
      const wrapper = shallow(
        <MapSearchResults
          isEmbed={false}
          location={{}}
          onItemClick={clickHandler}
          onMaximize={clickHandler}
          results={[]}
        />,
      )
      expect(wrapper).toMatchSnapshot()
    })

    it('should render results', () => {
      const count = 2
      const location = {
        latitude: 15.999,
        longitude: 329.123,
      }
      const clickHandler = jest.fn()
      const missingLayers = 'Layer 1, Layer 2'
      const results = [
        {
          label: 'label',
          categoryLabel: 'category',
          uri: 'result-uri-1',
          type: 'monumenten/monument',
          results: [],
          subCategories: [],
        },
        {
          label: 'label',
          categoryLabel: 'category',
          uri: 'result-uri-2',
          type: 'monumenten/monument',
          results: [],
          subCategories: [],
        },
      ]
      const wrapper = shallow(
        <MapSearchResults
          isEmbed={false}
          count={count}
          location={location}
          onItemClick={clickHandler}
          missingLayers={missingLayers}
          onMaximize={clickHandler}
          results={results}
        />,
      )
      expect(wrapper).toMatchSnapshot()
    })

    it('should filter external monumenten on pand results', () => {
      const count = 3
      const location = {
        latitude: 15.999,
        longitude: 329.123,
      }
      const clickHandler = jest.fn()
      const results = [
        {
          label: 'label',
          categoryLabel: 'category',
          uri: 'result-uri-1',
          type: 'bag/pand',
          results: [],
          subCategories: [],
        },
        {
          label: 'label',
          categoryLabel: 'category',
          uri: 'result-uri-2',
          type: 'foo',
          results: [],
          subCategories: [],
        },
        {
          label: 'label',
          categoryLabel: 'category',
          uri: 'result-uri-3',
          type: 'monumenten/monument',
          results: [],
          subCategories: [],
        },
      ]
      const wrapper = shallow(
        <MapSearchResults
          isEmbed={false}
          count={count}
          location={location}
          onItemClick={clickHandler}
          onMaximize={clickHandler}
          results={results}
        />,
      )
      expect(wrapper).toMatchSnapshot()
    })

    it('should update results', () => {
      const count = 2
      const location = {
        latitude: 15.999,
        longitude: 329.123,
      }
      const clickHandler = jest.fn()
      const results = [
        {
          label: 'label',
          categoryLabel: 'category',
          uri: 'result-uri-1',
          type: 'bag/pand',
          results: [],
          subCategories: [],
        },
        {
          label: 'label',
          categoryLabel: 'category',
          uri: 'result-uri-2',
          type: 'bag/pand',
          results: [],
          subCategories: [],
        },
      ]
      const wrapper = shallow(
        <MapSearchResults
          isEmbed={false}
          count={count}
          location={location}
          onItemClick={clickHandler}
          onMaximize={clickHandler}
          results={results}
        />,
      )
      expect(wrapper).toMatchSnapshot()

      wrapper.setProps({
        count: 3,
        location: {
          latitude: 1,
          longitude: 0,
        },
        panoUrl: 'other-pano-url',
        missingLayers: 'Layer 3',
        results: [
          {
            label: 'label',
            categoryLabel: 'category',
            uri: 'result-uri-1',
            type: 'bag/pand',
            results: [],
            subCategories: [],
          },
          {
            label: 'label',
            categoryLabel: 'category',
            uri: 'result-uri-2',
            type: 'bag/pand',
            results: [],
            subCategories: [],
          },
          {
            label: 'label',
            categoryLabel: 'category',
            uri: 'result-uri-3',
            type: 'bag/pand',
            results: [],
            subCategories: [],
          },
        ],
      })
      expect(wrapper).toMatchSnapshot()
    })
  })
})
