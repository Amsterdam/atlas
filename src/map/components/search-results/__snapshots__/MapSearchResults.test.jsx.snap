// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapSearchResults rendering should filter external monumenten on pand results 1`] = `
<section
  className="map-search-results"
>
  <header
    className="
          map-search-results__header
          map-search-results__header--no-pano
        "
  >
    <div
      className="map-search-results__header-container"
    >
      <h1
        className="map-search-results__header-title"
      >
        Resultaten (
        3
        )
      </h1>
      <h2
        className="map-search-results__header-subtitle"
      >
        locatie 
        1.12
        , 
        654.32
      </h2>
    </div>
  </header>
  <ul
    className="map-search-results__list"
  >
    <li
      key="result-uri-1"
    >
      <MapSearchResultsItem
        item={
          Object {
            "type": "bag/pand",
            "uri": "result-uri-1",
          }
        }
        onClick={[Function]}
      />
    </li>
    <li
      key="result-uri-2"
    >
      <MapSearchResultsItem
        item={
          Object {
            "type": "foo",
            "uri": "result-uri-2",
          }
        }
        onClick={[Function]}
      />
    </li>
  </ul>
</section>
`;

exports[`MapSearchResults rendering should render empty results 1`] = `
<section
  className="map-search-results"
>
  <header
    className="
          map-search-results__header
          map-search-results__header--no-pano
        "
  >
    <div
      className="map-search-results__header-container"
    >
      <h1
        className="map-search-results__header-title"
      >
        Resultaten (
        )
      </h1>
      <h2
        className="map-search-results__header-subtitle"
      >
        locatie 
        1.12
        , 
        654.32
      </h2>
    </div>
  </header>
  <ul
    className="map-search-results__list"
  />
</section>
`;

exports[`MapSearchResults rendering should render results 1`] = `
<section
  className="map-search-results"
>
  <header
    className="
          map-search-results__header
          map-search-results__header--pano
        "
  >
    <img
      alt="Panoramabeeld"
      className="map-search-results__header-pano"
      src="pano-url"
    />
    <div
      className="map-search-results__header-container"
    >
      <h1
        className="map-search-results__header-title"
      >
        Resultaten (
        2
        )
      </h1>
      <h2
        className="map-search-results__header-subtitle"
      >
        locatie 
        1.12
        , 
        654.32
      </h2>
    </div>
  </header>
  <ul
    className="map-search-results__list"
  >
    <li>
      <Notification
        className=""
        level="info"
      >
        Geen details beschikbaar van: 
        Layer 1, Layer 2
      </Notification>
    </li>
    <li
      key="result-uri-1"
    >
      <MapSearchResultsItem
        item={
          Object {
            "uri": "result-uri-1",
          }
        }
        onClick={[Function]}
      />
    </li>
    <li
      key="result-uri-2"
    >
      <MapSearchResultsItem
        item={
          Object {
            "uri": "result-uri-2",
          }
        }
        onClick={[Function]}
      />
    </li>
  </ul>
</section>
`;

exports[`MapSearchResults rendering should update results 1`] = `
<section
  className="map-search-results"
>
  <header
    className="
          map-search-results__header
          map-search-results__header--pano
        "
  >
    <img
      alt="Panoramabeeld"
      className="map-search-results__header-pano"
      src="pano-url"
    />
    <div
      className="map-search-results__header-container"
    >
      <h1
        className="map-search-results__header-title"
      >
        Resultaten (
        2
        )
      </h1>
      <h2
        className="map-search-results__header-subtitle"
      >
        locatie 
        1.12
        , 
        654.32
      </h2>
    </div>
  </header>
  <ul
    className="map-search-results__list"
  >
    <li
      key="result-uri-1"
    >
      <MapSearchResultsItem
        item={
          Object {
            "uri": "result-uri-1",
          }
        }
        onClick={[Function]}
      />
    </li>
    <li
      key="result-uri-2"
    >
      <MapSearchResultsItem
        item={
          Object {
            "uri": "result-uri-2",
          }
        }
        onClick={[Function]}
      />
    </li>
  </ul>
</section>
`;

exports[`MapSearchResults rendering should update results 2`] = `
<section
  className="map-search-results"
>
  <header
    className="
          map-search-results__header
          map-search-results__header--pano
        "
  >
    <img
      alt="Panoramabeeld"
      className="map-search-results__header-pano"
      src="other-pano-url"
    />
    <div
      className="map-search-results__header-container"
    >
      <h1
        className="map-search-results__header-title"
      >
        Resultaten (
        3
        )
      </h1>
      <h2
        className="map-search-results__header-subtitle"
      >
        locatie 
        1.12
        , 
        654.32
      </h2>
    </div>
  </header>
  <ul
    className="map-search-results__list"
  >
    <li>
      <Notification
        className=""
        level="info"
      >
        Geen details beschikbaar van: 
        Layer 3
      </Notification>
    </li>
    <li
      key="result-uri-1"
    >
      <MapSearchResultsItem
        item={
          Object {
            "uri": "result-uri-1",
          }
        }
        onClick={[Function]}
      />
    </li>
    <li
      key="result-uri-2"
    >
      <MapSearchResultsItem
        item={
          Object {
            "uri": "result-uri-2",
          }
        }
        onClick={[Function]}
      />
    </li>
    <li
      key="result-uri-3"
    >
      <MapSearchResultsItem
        item={
          Object {
            "uri": "result-uri-3",
          }
        }
        onClick={[Function]}
      />
    </li>
  </ul>
</section>
`;
